### PR TITLE
run build from setup.py for python apps

### PR DESCRIPTION
**Pull request by arroyc**
This PR is to support following scenarios:
1. if there is a setup.py file in the root level, and there is no requirement.txt, oryx should be able to detect it as python app and build smoothly.
2. if there is setup.py as well as requirement.txt, build from requirement.txt.

Note: we can also run build from both the files if both are present. I would like to know what's your thought on that.

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -68,6 +86,23 @@ then
...
+then
+	echo
+	START_TIME=$SECONDS
+	pip install --upgrade pip
+	ELAPSED_TIME=$(($SECONDS - $START_TIME))
+	echo "Done in $ELAPSED_TIME sec(s)."
+
+	echo "Running python setup.py install..."
+	$python setup.py install --user| ts $TS_FMT
+	cd *.egg-info
```
Comment (arroyc) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl]: python<version> setup.py install build and installs dependent packages. It installs packages in the userbase (it's needed for build image as in anteras  build env we don't have write permission) and also creates a <projectname>.egg-info folder. Inside that egg-info folder we have the requires.txt. which we do install again using pip (like we do for our normally when we see a requirement.txt in root of the app).

**Comment by arroyc** [comment]
python integration tests are successful .. some random node integration test failed/crashed

File: tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs [hunk 2]
```diff
@@ -400,6 +400,90 @@ public void GeneratesScript_AndBuilds_WhenExplicitLanguageAndVersion_AreProvided
...
+                result.GetDebugInfo());
+        }
+
+        // This is to test if we can build an app when both the files requirement.txt
+        // and setup.py are provided, we tend to prioritize the root level requirement.txt
+        [Fact]
+        public void GeneratesScript_AndBuilds_With_Both_Files_areProvided()
+        {
+            // Arrange
+            var appName = "flask-app";
```
Comment (kichalla) [inline tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs]: The test is talking about 'both' files being provided but this app only has `requirements.txt` file and not `setup.py`. Did you want to use `flask-setup-py-requirement-txt` app instead?

File: tests/Oryx.Integration.Tests/Python/PythonEndToEndTests.cs [hunk 3]
```diff
@@ -151,6 +151,44 @@ public async Task BuildWithVirtualEnv_RemovesOryxPackagesDir_FromOlderBuild(stri
                 });
         }
 
+        [Theory]
+        [InlineData("3.6")]
```
Comment (kichalla) [inline tests/Oryx.Integration.Tests/Python/PythonEndToEndTests.cs]: Do we need `2.7` too?

File: tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs (hunk 2 above)
Comment (arroyc) [reply tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs]: yes .. good catch … thanks !

**Pull request merged**